dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserService.java

```diff
@@ -330,9 +331,21 @@ private void handleUserQueryParams( UserQueryParams params )
             params.setInactiveSince( cal.getTime() );
         }
 
-        if ( params.isUserOrgUnits() && params.hasUser() )
+        if ( params.hasUser() )
         {
-            params.setOrganisationUnits( params.getUser().getOrganisationUnits() );
+            UserOrgUnitType orgUnitBoundary = params.getOrgUnitBoundary();
+            if ( params.isUserOrgUnits() || orgUnitBoundary == UserOrgUnitType.DATA_CAPTURE )
+            {
+                params.setOrganisationUnits( params.getUser().getOrganisationUnits() );
```

Comment:
I am thinking we could use just one collection, the one called `organisationUnits`, for all these boundaries. The type of org units can be inferred from the `UserOrgUnitType` in the SQL query. I might be wrong.

Reply:
yes, I think that should work. I'll update.